src/helpers/tests/artifacts/user.gen.ts

```diff
@@ -42,48 +51,80 @@ _id: mongoose.Types.ObjectId;
  * const userObject = user.toObject();
  * ```
  */
-export type User = {
+@Schema({"toObject":{"virtuals":true}})
+export class User extends mongoose.Types.Document {
+@Prop({"required":true,"type":String})
 email: string;
+@Prop({"required":true,"type":String})
 firstName: string;
 /** inline jsdoc */
+ @Prop({"required":true,"type":String})
 lastName: string;
 /**
  * single line jsdoc
  */
+ @Prop({"type":mongoose.Types.Mixed})
 metadata?: any;
+@Prop({"ref":"User","type":mongoose.Types.ObjectId})
 bestFriend?: User["_id"] | User;
 /**
  * multiline
  * jsdoc
  */
+ @Prop({"type":[UserFriendSchema],"required":true})
 friends: UserFriend[];
+@Prop({"type":"{\ncoordinates: Number[];\nsubdocWithoutDefault?: UserCitySubdocWithoutDefault[];\n}","required":true})
 city: {
 coordinates: number[];
 subdocWithoutDefault?: UserCitySubdocWithoutDefault[];
 };
+@Prop({"type":[String],"required":true})
 tags: string[];
+@Prop({"type":mongoose.Types.ObjectId})
 alternateObjectId?: mongoose.Types.ObjectId;
+@Prop({"type":mongoose.Types.Map<String>})
```

Comment:
Do you mean the type parameters of `Map` are different inside the `@Prop` and as the typescript type? If so, that's because `mongoose.Types.Map` always has keys of type String, so we only specify the type of the values, whereas for the typescript type we specify both.